### PR TITLE
add support for chef-zero/local-mode in client.rb

### PR DESCRIPTION
**Pull request by mburns**
fixes https://github.com/racker/chef/issues/4007

in `environments/performance.json`

```json
...
  "default_attributes": {
    "chef_client": {
      "local_mode": true
    }
  }
...
```

### CODE REVIEW

File: templates/default/client.rb.erb [hunk 1]
```diff
@@ -2,15 +2,28 @@ log_level        :info
 log_location     STDOUT
 log_location     open("/var/log/chef/client.log", "a")
 
+<% if node["chef_client"]["local_mode"] -%>
 chef_server_url  "<%= node["chef_client"]["server_url"] %>"
+<% else -%>
+chef_repo_path         "/home/mburns/chef"
```
Comment (drenalin23) [inline templates/default/client.rb.erb]: why do these have a hard coded path under your user homedir?

File: templates/default/client.rb.erb [hunk 2]
```diff
@@ -2,15 +2,28 @@ log_level        :info
...
+chef_zero.enabled      true
+local_mode             true
+
+environment_path '/home/mburns/chef/environments'
+node_path '/home/mburns/chef/nodes/maas-stage/'
+role_path '/home/mburns/chef/roles'
```
Comment (drenalin23) [inline templates/default/client.rb.erb]: same question as above.

File: templates/default/client.rb.erb (hunk 1 above)
Comment (mburns) [inline templates/default/client.rb.erb]: copypasta. I've updated so that `chef-zero` looks in `/data/chef/*` now

**Comment by drenalin23** [comment]
lgtm

**Comment by shawnashlee** [comment]
+1

**Comment by dimtruck** [comment]
🎉

**Pull request merged**